packages/notifi-react-card/lib/components/subscription/EventTypeHealthCheckRow.tsx

```diff
@@ -0,0 +1,209 @@
+import clsx from 'clsx';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { useNotifiSubscriptionContext } from '../../context';
+import { CheckRatio, HealthCheckEventTypeItem } from '../../hooks';
+import { DeepPartialReadonly } from '../../utils';
+import type { NotifiToggleProps } from './NotifiToggle';
+import { NotifiToggle } from './NotifiToggle';
+import { NotifiTooltip, NotifiTooltipProps } from './NotifiTooltip';
+
+export type EventTypeHealthCheckRowProps = Readonly<{
+  classNames?: DeepPartialReadonly<{
+    container: string;
+    label: string;
+    content: string;
+    button: string;
+    buttonContainer: string;
+    errorMessage: string;
+    toggle: NotifiToggleProps['classNames'];
+    tooltip: NotifiTooltipProps['classNames'];
+  }>;
+  disabled: boolean;
+  config: HealthCheckEventTypeItem;
+  inputs?: Record<string, string | undefined>;
+}>;
+
+export const EventTypeHealthCheckRow: React.FC<
+  EventTypeHealthCheckRowProps
+> = ({
+  classNames,
+  config,
+  disabled,
+  inputs,
+}: EventTypeHealthCheckRowProps) => {
+  const { alerts, loading } = useNotifiSubscriptionContext();
+  // const { instantSubscribe } = useNotifiSubscribe({
+  //   targetGroupName: 'Default',
+  // });
+  const [enabled, setEnabled] = useState(false);
+  const [ratios, setRatios] = useState<CheckRatio[]>([]);
+  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const [selectedRatio, setSelectedRatio] = useState<string>('');
+  const [customValue, setCustomValue] = useState<string>('');
+  const [cutomButtonPlaceholder, setCustomButtonPlaceholder] =
+    useState<string>('Custom');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isShowSelectButton, setIsShowSelectButton] = useState<boolean>(false);
+
+  const alertName = useMemo<string>(() => config.name, [config]);
+  const tooltipContent = config.tooltipContent;
+  useEffect(() => {
+    if (loading) {
+      return;
+    }
+    const hasAlert = alerts[alertName] !== undefined;
+    setEnabled(hasAlert);
+  }, [alertName, alerts, loading]);
+
+  useEffect(() => {
+    if (config && config.checkRatios.type === 'value') {
+      setRatios(config.checkRatios.value);
+    }
+  }, [config, setRatios]);
+
+  const handleNewSubscription = () => {
+    if (loading) {
+      return;
+    }
+    if (!isShowSelectButton) {
+      setIsShowSelectButton(true);
+      setEnabled(true);
+    } else {
+      const selectedRatioNumber = parseFloat(customValue.slice(0, -1));
+      if (
```

Comment:
Similar here, I think we should handle the check if the user clicks out of the input, not when they uncheck the input. 

I wonder if it would be better to also update NotifiToggle to allow passing a more general function as well. I ran into several issues toggling back and forth.

Currently, NotifiToggle accepts setChecked which is supposed to only update if it is checked or not:

```
onChange={(e) => {
          setChecked(e.target.checked);
        }}
```

I am not sure what the customer impact would be though!